src/main/java/seedu/address/logic/parser/FilterClientCommandParser.java

```diff
@@ -20,11 +23,25 @@ public class FilterClientCommandParser implements Parser<FilterClientCommand> {
     public FilterClientCommand parse(String args) throws ParseException {
         requireNonNull(args);
         ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_NAME);
-        String name = argMultimap.getValue(PREFIX_NAME).orElse("");
-        if (name.isEmpty()) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FilterClientCommand.MESSAGE_USAGE));
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_NAME)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FilterClientCommand.MESSAGE_USAGE));
         }
-        return new FilterClientCommand(new Name(name));
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME);
+        Name name = ParserUtil.parseClientName(argMultimap.getValue(PREFIX_NAME).get());
+        return new FilterClientCommand(name);
+    }
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     *
+     * @param argumentMultimap The argument multimap that holds the parsed arguments.
+     * @param prefixes The prefixes to check for presence.
+     * @return True if all prefixes contain non-empty values, false otherwise.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
+        return Stream.of(prefixes).allMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
```

Comment:
Great use of javadocs to explain methods whose behaviour and purpose may not be very apparent on first glance